icicle/backend/cpu/src/curve/cpu_msm.hpp

```diff
@@ -304,18 +354,20 @@ class Msm
 
 template <typename A, typename P>
 Msm<A, P>::Msm(const MSMConfig& config, const int c, const int nof_threads)
-    : manager(nof_threads),
+    : manager(nof_threads), m_curr_task(nullptr),
 
       m_c(c), m_num_bkts(1 << (m_c - 1)), m_precompute_factor(config.precompute_factor),
-      m_num_bms(((scalar_t::NBITS) / (config.precompute_factor * m_c)) + 1),
+      m_num_bms(((scalar_t::NBITS - 1) / (config.precompute_factor * m_c)) + 1),
       m_are_scalars_mont(config.are_scalars_montgomery_form), m_are_points_mont(config.are_points_montgomery_form),
       m_batch_size(config.batch_size),
 
       m_buckets(m_num_bms * m_num_bkts), m_bkts_occupancy(m_num_bms * m_num_bkts, false),
 
       m_log_num_segments(std::max(
-        (int)std::floor(std::log2((double)(nof_threads * TASKS_PER_THREAD - 1) / (double)(2 * m_num_bms))), 0)),
-      m_num_bm_segments(std::min((int)(1 << m_log_num_segments), (int)(m_num_bms * m_num_bkts))),
+        (int)std::floor(
+          std::log2((double)(nof_threads * TASKS_PER_THREAD * EC_BATCH_SIZE - 1) / (double)(2 * m_num_bms))),
```

Comment:
do you really need the double casting?

Reply:
I can remove one of the (double)s if that is what you mean.